Wikipedia/Code/WMFImageURLParsing.m

```diff
@@ -6,7 +7,7 @@
     dispatch_once(&onceToken, ^{
         // TODO: try to read serialized regex from disk to prevent needless pattern compilation on next app run
         NSError* patternCompilationError;
-        imageNameFromURLRegex = [NSRegularExpression regularExpressionWithPattern:@"^\\d+px-([^.]*(\\.[^.$]{4,})*\\.[^.]*).*$"
+        imageNameFromURLRegex = [NSRegularExpression regularExpressionWithPattern:@"^\\d+px-(.*)"
```

Comment:
@niedzielski in our case, the lead image was fine, it was the one after it that was problematic. See the example URL in the test case I linked to